buildsystem: Fully rework

robimarko opened the pull request:
Completely rework the current build system.
Switch from building from source to using the upstream provided imagebuilder.

Use upstream SDK to build not yet upstreamed packages like tunneldigger.

This change is drop in compatible with Nodewatcher as the same image type is generated at the end.

Both base and runtime has been updated to use Ubuntu 18.04.

There are still improvements to be made in the following time.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
docker: Update base image to Ubuntu 18.04 Bionic

Update the base image used for image building to Ubuntu 18.04 Bionic.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
Drop legacy OpenWrt

Drop the legacy and now unused OpenWrt directory and accompanying stuff.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
lede: Rebrand to OpenWrt

Rebrand what is currently LEDE to OpenWrt since those two merged and are now OpenWrt once again.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
openwrt: Add build script

Initial commit for build script

robimarko committed:
buildsystem: Fully rework

Completely rework the current build system.
Switch from building from source to using the upstream provided imagebuilder.

Use upstream SDK to build not yet upstreamed packages like tunneldigger.

This change is drop in compatible with Nodewatcher as the same image type is generated at the end.

Both base and runtime has been updated to use Ubuntu 18.04.

There are still improvements to be made in the following time.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
docs: Update to current

Update the docs to match the current image building procedure.

kostko reviewed (approved):
Nice work! I did a quick pass and it seems great, simplification is always nice! :-)

robimarko commented:
Thanks, it's still not fully complete.
I am having issues getting the package2json working.
That's the last part, but I am sure some bugs will pop up.

Next big step is to tackle packages

robimarko committed:
tools: packages2json: Bypass delimiter issue

Discard lines that have multiple delimiters, otherwise, no metadata will be written.
This previously was not an issue but the introduction of kmod-dax package in late 2017 causes multiple delimiters to be found in the same line and split function does not like that.
https://github.com/openwrt/openwrt/commit/38be62b97ecd313b725f1165a4ffedf166095c06
As you can see Title which is later translated to Description field also has ": " in it causing multiple delimiters to be found.
So simply discard any line that this occurs to, this is not a big issue since Description is not used at all.

robimarko committed:
firmware-runtime: Add bsdmainutils

Add bsdmainutils to the firmware-runtime docker images.
Imagebuilder complains about it missing for some functions.

robimarko committed:
openwrt: FIx package copy step

Previous method was obviously wrong.
It copied only kmods as generated packages are in bin/packages.
So use find to find them and xargs with cp to copy them properly.

robimarko committed:
openwrt: Mark imagebuilder as LEDE in metadata

Mark imagebuilders as LEDE until Nodewatcher is updated to merge LEDE and OpenWrt there.
Otherwise configuration is not correct.

robimarko committed:
openwrt: Fix ownership issues

As the imagebuilder is copied later that the directory is chowned it remains under the root user and Nodewatcher cannot later clean up the build remains.

robimarko committed:
openwrt: Add other Nodewatcher required packages

Add other packages that Nodewatcher needs to be compiled

robimarko commented:
So this is pretty much it.
I confirmed it works and generates images in Nodewatcher.
The only downside is that Docker images will be larger as OpenWrt requires the full package stack as it requires when building from source.
